WebProcessProxy should keep WebsiteDataStore alive

achristensen07 opened the pull request:
#### 2cd7eec58eff04dacbc4466941627d044f901adf
<pre>
WebProcessProxy should keep WebsiteDataStore alive
<a href="https://bugs.webkit.org/show_bug.cgi?id=243960">https://bugs.webkit.org/show_bug.cgi?id=243960</a>
&lt;rdar://98313174&gt;

Reviewed by Chris Dumez.

This effectively reverts bug 238892 because it caused rdar://98313174
Tasks such as the one scheduled in WorkerSWClientConnection::unregisterServiceWorkerClient
need a connection to the network process, and if the network process has crashed
then they need a WebsiteDataStore to restart that network process, so a WebProcessProxy
needs to keep a WebsiteDataStore alive.

* Source/WebKit/UIProcess/WebProcessProxy.cpp:
(WebKit::m_webPermissionController):
(WebKit::WebProcessProxy::setWebsiteDataStore):
(WebKit::WebProcessProxy::isDummyProcessProxy const):
(WebKit::WebProcessProxy::addExistingWebPage):
(WebKit::WebProcessProxy::getNetworkProcessConnection):
(WebKit::WebProcessProxy::sessionID const):
(WebKit::WebProcessProxy::websiteDataStore const): Deleted.
* Source/WebKit/UIProcess/WebProcessProxy.h:
(WebKit::WebProcessProxy::websiteDataStore const):

Canonical link: <a href="https://commits.webkit.org/253449@main">https://commits.webkit.org/253449@main</a>
</pre>

cdumez commented on Source/WebKit/UIProcess/WebProcessProxy.cpp:
```diff
@@ -802,9 +793,7 @@ void WebProcessProxy::updateBackForwardItem(const BackForwardListItemState& item
-    if (!dataStore)
-        return reply({ });
-
+    RELEASE_ASSERT(dataStore);
```
This looks very dangerous. If this happened, we'd be converting a WebContent process crash into a UIProcess crash, which is not a good trade off IMO. I'd recommend a debug ASSERT() instead and a RELEASE_LOG_FAULT() so we know it's happening.

cdumez reviewed (approved):
LGTM with suggested change.

achristensen07 committed:
WebProcessProxy should keep WebsiteDataStore alive
https://bugs.webkit.org/show_bug.cgi?id=243960
<rdar://98313174>

Reviewed by Chris Dumez.

This effectively reverts bug 238892 because it caused rdar://98313174
Tasks such as the one scheduled in WorkerSWClientConnection::unregisterServiceWorkerClient
need a connection to the network process, and if the network process has crashed
then they need a WebsiteDataStore to restart that network process, so a WebProcessProxy
needs to keep a WebsiteDataStore alive.

* Source/WebKit/UIProcess/WebProcessProxy.cpp:
(WebKit::m_webPermissionController):
(WebKit::WebProcessProxy::setWebsiteDataStore):
(WebKit::WebProcessProxy::isDummyProcessProxy const):
(WebKit::WebProcessProxy::addExistingWebPage):
(WebKit::WebProcessProxy::getNetworkProcessConnection):
(WebKit::WebProcessProxy::sessionID const):
(WebKit::WebProcessProxy::websiteDataStore const): Deleted.
* Source/WebKit/UIProcess/WebProcessProxy.h:
(WebKit::WebProcessProxy::websiteDataStore const):

Canonical link: https://commits.webkit.org/253449@main
